app/vmalert/group.go

```diff
@@ -50,6 +50,17 @@ func newGroup(cfg config.Group, defaultInterval time.Duration) *Group {
 	}
 	rules := make([]Rule, len(cfg.Rules))
 	for i, r := range cfg.Rules {
+		// override rule labels with external labels
+		for k, v := range labels {
+			if prevV, ok := r.Labels[k]; ok {
+				logger.Infof("label %q=%q for rule %q.%q overwritten with external label %q=%q",
```

Comment:
in prometheus external label has not effect when metrics has the same label key 

```
# Note that any globally configured "external_labels" are unaffected by this # setting. In communication with external systems, they are always applied only # when a time series does not have a given label yet and are ignored otherwise.
```

Reply:
True! I wonder what are the reasons to respect original label instead of renaming it. Btw, looks like vmagent does not respect original label as well - https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmagent/remotewrite/relabel.go#L91
@valyala

Reply:
I think this behavior must be fixed in `vmagent` in order to be consistent with Prometheus.

Reply:
Agreed